lib/jsonapi/authorization/default_pundit_authorizer.rb

```diff
@@ -245,6 +292,23 @@ def authorize_relationship_operation(source_record, relationship_method, *args)
           ::Pundit.authorize(user, source_record, 'update?')
         end
       end
+
+      def relationship_method(data, **options)
+        relationship = data[:relationship]
+        assoc_name   = data[:relation_name]
+        prefix       = options[:prefix]
+
+        case relationship
+        when ->(relationship) { relationship.polymorphic }
+          polymorphic_type = data[:records].class.name.downcase
+          "#{prefix}_#{relationship.class_name.downcase}_#{polymorphic_type}?"
+        when ->(relationship) { relationship.class == JSONAPI::Relationship::ToOne }
```

Comment:
In order to call a method on the case object (like `.polymorphic` and `.class`) I have to do this via a lambda. `when relationship.polymorphic` and `relationship.class == JSONAPI::Relationship::HasOne` always come back false even when they should be true. Perhaps I am missing something here? Please let me know. It only works properly for me when using lambdas.